src/libraries/System.Private.Xml.Linq/src/System/Xml/Linq/XDocumentType.cs

```diff
@@ -53,20 +54,17 @@ internal XDocumentType(XmlReader r)
         /// <summary>
         /// Gets or sets the internal subset for this Document Type Definition (DTD).
         /// </summary>
+        [AllowNull]
         public string InternalSubset
         {
             get
             {
-                // TODO-NULLABLE: As per documentation, this should return string.Empty.
-                // Should we check for null here?
-                // This is also referenced by XNodeReader.Value which overrides XmlReader.Value, which is non-nullable.
-                // There is one case that passes a nullable parameter (XNodeBuilder.WriteDocType), currently we are just asserting that the nullable parameter does not receive null.
                 return _internalSubset;
             }
             set
             {
                 bool notify = NotifyChanging(this, XObjectChangeEventArgs.Value);
-                _internalSubset = value;
+                _internalSubset = value ?? string.Empty;
```

Comment:
Should all the null checks related to _internalSubset be removed now?

Reply:
there is just one in GetHashCode, will remove